src/main/java/org/sqlite/core/NativeDB.c

```diff
@@ -474,11 +490,23 @@ JNIEXPORT jint JNICALL JNI_OnLoad(JavaVM *vm, void *reserved)
     phandleclass = (*env)->FindClass(env, "org/sqlite/ProgressHandler");
     if(!phandleclass) return JNI_ERR;
     phandleclass = (*env)->NewWeakGlobalRef(env, phandleclass);
+    phandle_mth_progress = (*env)->GetMethodID(env, phandleclass, "progress", "()I");
 
-    jclass exclass = (*env)->FindClass(env, "java/lang/Throwable");
+    bhandleclass = (*env)->FindClass(env, "org/sqlite/BusyHandler");
+    if(!bhandleclass) return JNI_ERR;
+    bhandleclass = (*env)->NewWeakGlobalRef(env, bhandleclass);
+    bhandle_mth_callback = (*env)->GetMethodID(env, bhandleclass, "callback", "(I)I");
+
+    exclass = (*env)->FindClass(env, "java/lang/Throwable");
+    if(!exclass) return JNI_ERR;
+    exclass = (*env)->NewWeakGlobalRef(env, exclass);
```

Comment:
Lack of global reference to this class caused the Segfault mentioned in the surefire exclusions. The class reference got garbage collected which invalidates the jmethodID which caused usage of freed memory.

Reply:
It seems bizarre to me that java.lang.Throwable can be unloaded when we instantiate an Exception where Throwable is an interface.

Also, since this is a call to `toString()`, I think that we can use java.lang.Object, rather than Throwable.

Also, as we're holding a weak global reference, I'm confused at what would be stopping the VM from unloading the class fully, as a weak reference shouldn't prevent this.

With this said, I am not super familiar with these APIs.

Reply:
Looking into this more, I think what we ran into is this: https://stackoverflow.com/questions/11907832/jni-local-vs-global-reference-is-not-a-valid-jni-reference

It appears we're just lucky that the regular JVM isn't moving these classes around.

It seems that the proper fix is that these should be all global references (these classes should not be unloaded), or FindClass should be called in the places where these classes are used. Calling FindClass more often is likely cleaner, but allows for more runtime exceptions due to graal native image misconfiguration and may be slower.